### ISSUE TITLE
[Bug]: STSWebIdentityAWSCredentialIdentityResolver not working correctly

### ISSUE THREAD

**Issue by shepazon** [awslabs/aws-sdk-swift#1694]
### Describe the bug

When trying to create an `STSWebIdentityAWSCredentialIdentityResolver` with a JWT returned by Sign In With Apple, the error below occurs:
```
crtError(AwsCommonRuntimeKit.CRTError(code: 6160, message: "Valid credentials could not be sourced by the sts web identity provider", name: "AWS_AUTH_CREDENTIALS_PROVIDER_STS_WEB_IDENTITY_SOURCE_FAILURE"))
```

Sebastien has reproduced this issue with an independently created program.

This is the sample program I created that encounters this issue (though I've tried others as well): [Buckets.zip](https://github.com/user-attachments/files/16749984/Buckets.zip)

### Expected Behavior

When trying to create an `STSWebIdentityAWSCredentialIdentityResolver` with a JWT returned by Sign In With Apple, an identity resolver that returns valiid temporary AWS credentials should be returned (assuming that both SIWA and AWS are correctly configured).

### Current Behavior

See "describe the bug."

### Reproduction Steps

1. Create a program that uses Sign In With Apple to get a JWT, then uses it to create an STSWebIdentityAWSCredentialIdentityResolver.
2. Make sure the program has the SIWA entitlement.
3. Use Apple's developer portal to set up an app that uses SIWA to return a token, with the app's bundle ID used as the audience for the token.
4. Add appleid.apple.com as an identity source for IAM and configure it to associate the audience with the role that the app should be using.
5. Run the app. If everything is configured right, the error above will occur, even though it should actually correctly gain AWS access.

### AWS SWIFT SDK version used

0.65.0

### Compiler and Version used

Xcode 15.4

### Operating System and version

Sonoma 14.6.1

**Comment by sichanyoo** [comment]
Created a ticket in downstream repo aws-crt-swift: https://github.com/awslabs/aws-crt-swift/issues/279

**Comment by sebsto** [comment]
In my tests, I can reproduce this. 
i checked the token returned by SIWA is valid (by copy / pasting it to jwt.io and manually looking at the results) 
I also checked the IAM configuration is correct. When I write code to call STS AssumeRoleWithWebIdentity, it correctly returns an access key and secret key

**Comment by sichanyoo** [comment]
The bug turned out to be due to the provided role session name not following the regex outlined in the docs [here](https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRoleWithWebIdentity.html). The role session name that was used with this sample project had a white space in it.

After discussion with @waahm7, as CRT does no validation of input parameters and C does not have regex parser, we decided it would be best if we add this input parameter check to SDK wrapper. 

Closing this ticket (and the one in downstream CRT repo) as it is not an issue with the SDK / CRT.
I'll be creating a follow-up ticket to this for adding role session name param check in our resolve wrappers.
